### ISSUE TITLE
lambda function in class raises incompatible error

### ISSUE THREAD

**Issue by williamfzc** [microsoft/pyright#366]
**Describe the bug**

I think code is better :)

```python
import typing


class SampleObject:
    def __init__(self, inner_str: str):
        self.inner_str: str = inner_str

    def a(self, s: str):
        return self.inner_str + s

    b: typing.Callable[["SampleObject", str], str] = lambda a_inst, s: a_inst.inner_str + s


sample = SampleObject("this is ")
test_string = "a string"
print(sample.a(test_string))
print(sample.b(test_string))
```

output:

```
this is a string
this is a string
```

it works fine. and run pyright:

```
  17:7 - error: Argument missing for parameter 'p1'
  17:16 - error: Argument of type 'Literal['a string']' cannot be assigned to parameter 'p0' of type 'SampleObject'
  'str' is incompatible with 'SampleObject'
2 errors, 0 warnings 
Completed in 1.279sec
```

and if I edit b like this:

```python
b: typing.Callable[[str], str] = lambda a_inst, s: a_inst.inner_str + s
```

it also raised an error:

```
  11:63 - error: Cannot access member 'inner_str' for type 'str'
  Member 'inner_str' is unknown
  11:38 - error: Expression of type '(a_inst: str, s: Unknown) -> Unknown' cannot be assigned to declared type '(p0: str) -> str'
  Function accepts too few parameters. Expected 2 but got 1.
2 errors, 0 warnings 
Completed in 1.262sec
```

**Expected behavior**

maybe it should not raise an error here.
or, is there a better way to write the typing hints for it?

pyright 1.1.5

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem, and I'll work on a fix.

This hasn't come up previously because using a lambda in this context is very unusual. Normally, a function declaration would be used to specify a class member.

```
def b(self, s: str):
   return self.inner_str + s
```

I'm curious why you have used a lambda in this case.

**Comment by williamfzc** [comment]
```
Normally, a function declaration would be used to specify a class member.
```

yep, I agree.

```
I'm curious why you have used a lambda in this case.
```

just a casual try when I writing a very short function, not a real practice in production :)

**Comment by erictraut** [comment]
This will be fixed in the next version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.1.14, which I just published.
